x/exchange/client/cli/query_setup.go

```diff
@@ -341,6 +342,9 @@ func MakeQueryGetAccountCommitments(_ client.Context, flagSet *pflag.FlagSet, ar
 
 	var err error
 	rv.Account, err = ReadStringFlagOrArg(flagSet, args, FlagAccount, "account")
+	if asset, err := flagSet.GetString("asset"); err == nil && asset != "" {
+		rv.XAsset = &exchange.QueryGetAccountCommitmentsRequest_Asset{Asset: asset}
+	}
 
 	return rv, err
```

Comment:
One thing to watch for in go is that errors are being properly checked. With your change, the `err` from reading the account (line 344) is now being ignored. Usually, we'd handle that with an `if err != nil` check right after, but in these makers, we read all the fields and return multiple errors if needed. So instead of all those `if err != nil` checks, we create a `[]error` and put each field's error in a specific index and, at the end, combine them all using `errors.Join(...)` (which ignores nil entries). See `MakeQueryGetMarketCommitments` for an example.

Once you remove the `optional` keyword from the proto field, this'll be easier too.

```suggestion
	errs := make([]error, 2)
	rv.Account, errs[0] = ReadStringFlagOrArg(flagSet, args, FlagAccount, "account")
	rv.Denom, errs[1] = flagSet.GetString(FlagDenom)

	return rv, errors.Join(errs...)
```